triage/experiments/validate.py

```diff
@@ -0,0 +1,221 @@
+            if not hasattr(value, '__iter__') or isinstance(value, (str, bytes)):
+                raise ValueError('Each value in FeatureGroupCreator must be iterable and not a string')
+
+        if 'prefix' in feature_group_definition:
+            available_prefixes = [
+                aggregation['prefix']
+                for aggregation in feature_aggregation_config
+            ]
+            for prefix in feature_group_definition['prefix']:
+                if prefix not in available_prefixes:
+                    raise ValueError('''Aggregation prefix of '{}' was given as a
+                    feature group, but no such prefix exists in the available
+                    feature aggregations. The available feature aggregations are {}
+                    '''.format(prefix, available_prefixes))
+
+class FeatureGroupStrategyValidator(Validator):
+    def run(self, feature_group_strategies):
+        if not isinstance(feature_group_strategies, list):
+            raise ValueError('feature_group_strategies key must be a list')
+        available_strategies = architect.feature_group_mixer.FeatureGroupMixer.strategy_lookup
+        for strategy in feature_group_strategies:
+            if strategy not in available_strategies:
+                raise ValueError('''Unknown feature_group_strategies key {} received.
+                Available keys are {}'''.format(strategy, available_strategies))
+
+
+class UserMetadataValidator(Validator):
+    def run(self, user_metadata):
+        if not isinstance(user_metadata, dict):
+            raise ValueError('user_metadata key must be a dict')
+
+
+class ModelGroupKeysValidator(Validator):
+    def run(self, model_group_keys, user_metadata):
+        if not isinstance(model_group_keys, list):
+            raise ValueError('model_group_keys must be a list')
+        # planner_keys are defined in architect.Planner._make_metadata
+        planner_keys = [
+            'beginning_of_time',
+            'end_time',
+            'indices',
+            'feature_names',
+            'label_name',
+            'label_type',
+            'state',
+            'matrix_id',
+            'matrix_type'
+        ]
+        # temporal_keys are defined in timechop.Timechop.generate_matrix_definition
+        temporal_keys = [
+            'matrix_start_time',
+            'matrix_end_time',
+            'as_of_times',
+            'label_window',
+            'example_frequency',
+            'train_duration'
+        ]
+        available_keys = [key for key in user_metadata.keys()] + planner_keys + temporal_keys
+        for model_group_key in model_group_keys:
+            if model_group_key not in available_keys:
+                raise ValueError('''Unknown model_group_keys entry '{}' received.
+                Available keys are {}'''.format(model_group_key, available_keys))
+
+
+class GridConfigValidator(Validator):
+    def run(self, grid_config):
+        pass
+
+
+class ScoringConfigValidator(Validator):
```

Comment:
At least ensure that the specified metrics exist and are properly defined with a metric decorator to specify whether greater is better?